ci-operator/step-registry/bastion/dnsmasq/rhcos/bastion-dnsmasq-rhcos-ref.yaml

```diff
@@ -0,0 +1,14 @@
+ref:
+  as: bastion-dnsmasq-rhcos
```

Comment:
@jinyunma, is it possible to merge `bastion-dnsmasq` and `bastion-dnsmasq-rhcos`? we can introduce a flag like `ENABLE_DNSMASQ_ON=(Service|NetworkManager)`, WDYT?

Reply:
hmm, I can do that. 
I'm a little confused about the step function, when developing azure-provision-customer-dns, it is suggested to split into three steps for mixed, public and private so that we can easy to maintain in the future. Actually most codes are duplicated, we only need to write dns records entry into different files in shared dir based on publish strategy defined in install-config.
so with same consideration in mind, I created new step to configure dnsmasq in NetworkManager, since this step only used on Azure platform so far.